hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -1799,10 +1800,14 @@ public void satisfyStoragePolicy(String path) throws IOException {
   }
 
   @Override
-  public HAServiceProtocol.HAServiceState getHAServiceState()
-      throws IOException {
-    rpcServer.checkOperation(NameNode.OperationCategory.READ, false);
-    return null;
+  public HAServiceProtocol.HAServiceState getHAServiceState() {
+    try {
+      // check safemode or not
+      rpcServer.checkOperation(NameNode.OperationCategory.WRITE);
```

Comment:
Instead of using exceptions, can we extract the code from checkSafeMode() so we can just check the state?

Reply:
Thank you for your comment. Updated the patch to just check the state instead of using StandbyException.